dataland-frontend/src/components/forms/parts/elements/derived/ProcurementCategoryFormElement.vue

```diff
@@ -0,0 +1,177 @@
+            </div>
+          </FormKit>
+        </div>
+      </div>
+    </FormKit>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import InputSwitch from "primevue/inputswitch";
+import { BaseFormFieldProps } from "@/components/forms/parts/fields/FormFieldProps";
+import { FormKit } from "@formkit/vue";
+import UploadFormHeader from "@/components/forms/parts/elements/basic/UploadFormHeader.vue";
+import MultiSelectFormFieldBindData from "@/components/forms/parts/fields/MultiSelectFormFieldBindData.vue";
+import PrimeButton from "primevue/button";
+import NaceCodeFormFieldBindData from "@/components/forms/parts/fields/NaceCodeFormFieldBindData.vue";
+import { DropdownDatasetIdentifier, getDataset } from "@/utils/PremadeDropdownDatasets";
+import { LksgProcurementCategory } from "@clients/backend";
+
+export default defineComponent({
+  name: "ProcurementCategoryFormElement",
+  inject: {
+    procurementCategories: {
+      from: "procurementCategories",
+      default: {} as { [key: string]: LksgProcurementCategory },
+    },
+  },
+  components: {
+    InputSwitch,
+    FormKit,
+    UploadFormHeader,
+    MultiSelectFormFieldBindData,
+    PrimeButton,
+    NaceCodeFormFieldBindData,
+  },
+  props: BaseFormFieldProps,
+  data() {
+    return {
+      isItActive: !!this.procurementCategories[this.name],
+      procuredProductTypesAndServicesNaceCodesValue: [],
+      percentageOfTotalProcurementValue: "",
+      allCountry: getDataset(DropdownDatasetIdentifier.CountryCodes),
+      selectedCountries: [],
+      numberOfSuppliersPerCountryCodeValue: [],
+    };
+  },
+  watch: {
+    preSelectedCountries(newValue: []) {
+      this.selectedCountries = newValue;
+    },
+  },
+  computed: {
+    preSelectedCountries() {
+      return this.allCountry.filter((el) =>
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access,no-prototype-builtins
+        this.procurementCategories[this.name]?.numberOfSuppliersPerCountryCode?.hasOwnProperty(el.value)
+      );
+    },
+  },
+  methods: {
+    /**
+     * Remove item from selected countries list
+     * @param countryCode - Country code to be removed
+     */
+    removeItemFromListOfSelectedCountries(countryCode: string) {
+      this.selectedCountries = this.selectedCountries.filter((el) => countryCode !== el.value);
+    },
+
+    /**
+     * remove the word from the phrase
+     * @param word - word to remove
+     * @param phrase - phrase from which we remove
+     * @returns phrase without the removed word
+     */
+    removeWordFromPhrase(word: string, phrase: string): string {
+      const splittedPhrase: [string] = phrase.split(" ");
+      for (let i = splittedPhrase.length - 1; i >= 0; i--) {
+        if (splittedPhrase[i].includes(word)) {
+          splittedPhrase.splice(i, 1);
+          break;
+        }
+      }
+      return splittedPhrase.join(" ");
+    },
+  },
+});
```

Comment:
This method should not be here in my opinion.

I think your goal is to filter out the country name from phrases like "Albania (AL)".   So
"Albania (AL)" should be just "Albania"

I suggest

a) To write a function for this directly in the file "PremadeDropddownDatasets", where all the country code stuff resides.

b) I would call the function something like `getCountryNameFromCountryCode( countryCode: string ) {...}`

c) In the function itself, I would simply `getAllCountryNamesWithCodes()` and then pick the country name based on the country code.

I know this is probably a little more computing than it is now.  But it is client side code so I think it is acceptable.

Reply:
I just noticed that we already have a function `getCountryNameFromCountryCode(countryCode: string)` in our file `CountryCodeConverter`

Reply:
👍